h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/XGBoostUtils.java

```diff
@@ -388,7 +387,7 @@ private static long denseChunk(float[][] data,
                     currentCol += offset;
                 }
                 for (int j = 0; j < di._nums; ++j) {
-                    if(currentCol == ARRAY_MAX) {
+                    if(currentCol >= ARRAY_MAX) {
```

Comment:
@wendycwong Just a question, did we get and overflow here ?

Reply:
Pavel:  Not sure.  But I was just being paranoid since if the case currentCol==ARRAY_MAX imply that currentCol>=ARRAY_MAX is true but the reverse is not true.  Also, there are cases when currentCol > ARRAY_MAX might happen.  Hence, this one will catch everything.